HelloFlask/views.py

```diff
@@ -1,31 +1,44 @@
-from flask import Flask
-from flask import render_template
+from datetime import datetime
+from re import match
+
+from flask import Flask, render_template
+
 from HelloFlask import app
 
+
 @app.route('/')
 def home():
-    return render_template("home.html", title = "Home")
+    return render_template("home.html", title="Home")
 
 @app.route('/about')
 def about():
-    return render_template("about.html", title = "About us")
+    return render_template("about.html", title="About us")
 
 @app.route('/contact')
 def contact():
-    return render_template("contact.html", title = "Contact us")
+    return render_template("contact.html", title="Contact us")
 
 @app.route('/hello/<name>')
 def hello_there(name):
-    from datetime import datetime
-    now = datetime.now()
-
     return render_template(
         "hello_there.html",
         title ='Hello, Flask',
-        message = "Hello there, " + name + "!",
-        date = now.strftime("%A, %d %B, %Y at %X")
+        name = clean_name(name),
+        date = datetime.now()
     )
 
 @app.route('/api/data')
 def get_data():
-    return app.send_static_file('data.json')
\ No newline at end of file
+    return app.send_static_file('data.json')
+
+def clean_name(name):
```

Comment:
this is not really needed, since any variable passed to jinja is considered unsafe by default so any HTML there will be escaped

Reply:
Of course...I had it there in an intermediate step of the tutorial when I was just generating a plain text response, so I can omit it from the final version. Thanks for the reminder.